Sprint-2/errors/2.js

```diff
@@ -3,8 +3,35 @@
 
 // this function should square any number but instead we're going to get an error
 
-function square(3) {
+function square(num) {
     return num * num;
 }
 
 
+console.log(`square for 2 is ${square(2)}`);
+console.log(`square for 3 is ${square(3)}`);
+console.log(`square for 4 is ${square(4)}`);
+
+
+/* My predictions:
+
+I hope everything is simple and obvious here. 
+Somehow, the exercises went from complex to simple, 
+usually the other way around.
+
+Here the error will be due to the fact that 
+the identifier of the square() function parameter 
```

Comment:
Why do you think it cannot start with a number?

Reply:
I don't think, I know and I wrote about it. 😊

The Identifier name cannot start with a number. This is a Rule or axiom of any programming language, this rule was thought up by the creators of the first programming language and this rule is followed by all popular languages ​​that were created and are created after that

Reply:
So, another way to say this might be, is there a difference between this cannot be "of the data type number" and this cannot be "a string that begins with a character that represents a number" ? Would the following produce different or the same errors? (And why)

```
function num(3_set) { return true}
function num(3set) { return true}
function num(3) { return true}
```

Reply:
Sorry, maybe I'm not expressing myself clearly. English is not my native language. But I'll just try to repeat it again, I don't know any other answer. Identifiers cannot be specified starting with a number in any programming language I know. This is the rule for assigning names to identifiers and all your 3 lines:
```
function num(3_set) { return true}
function num(3set) { return true}
function num(3) { return true}
```
should give a syntax error, because all three identifiers start with a number. I wrote about this both in the comment to the task and in our discussion. To avoid errors, you need to assign other identifiers to the parameters of all your functions. For example, these:
```
function num(_3_set) { return true}
function num(_3set) { return true}
function num(_3) { return true}
```